src/sentry/incidents/models.py

```diff
@@ -623,6 +623,30 @@ def get_registered_types(cls):
         return cls._type_registrations.values()
 
 
+class AlertRuleActivityType(Enum):
+    CREATED = 1
+    DELETED = 2
+    UPDATED = 3
+    ENABLED = 4
+    DISABLED = 5
+
+
+class AlertRuleActivity(Model):
+    __core__ = True
+
+    alert_rule = FlexibleForeignKey("sentry.AlertRule")
+    previous_alert_rule = FlexibleForeignKey(
```

Comment:
When does this column get used?

Reply:
Whenever an alert rule is changed, and it has associated incidents/alerts, we create a new instance of the rule and archive the old one (the old one remains the same, and the newly created one reflects their most recent edits). In those cases, I thought storing a reference to the rule it "came from" could be helpful down the line.

Reply:
Makes sense to me.